component/xload/xload.go

```diff
@@ -466,6 +467,93 @@ func (xl *Xload) CloseFile(options internal.CloseFileOptions) error {
 	return nil
 }
 
+// Read-only mode enforcement: Override write operations to return EROFS
+
+func (xl *Xload) CreateFile(options internal.CreateFileOptions) (*handlemap.Handle, error) {
```

Comment:
When xload component is plugged in the pipeline, it mount the file-system in "ro" mode where kernel shall not even allow any write file operations on the mount path. We need not implement all these methods as kernel shall be blocking all such operations.